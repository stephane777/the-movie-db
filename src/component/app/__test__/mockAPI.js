export const resultMockAPI = [
	{
		adult: false,
		backdrop_path: "/jMWkd0fuwbG39eJpzycJzPWMCww.jpg",
		genre_ids: [28, 878],
		id: 399566,
		original_language: "en",
		original_title: "Godzilla vs. Kong",
		overview:
			"In a time when monsters walk the Earth, humanity’s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.",
		popularity: 12276.164,
		poster_path: "/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg",
		release_date: "2021-03-24",
		title: "Godzilla vs. Kong",
		video: false,
		vote_average: 8.5,
		vote_count: 3719,
	},
	{
		adult: false,
		backdrop_path: "/hJuDvwzS0SPlsE6MNFOpznQltDZ.jpg",
		genre_ids: [16, 12, 14, 10751, 28],
		id: 527774,
		original_language: "en",
		original_title: "Raya and the Last Dragon",
		overview:
			"Long ago, in the fantasy world of Kumandra, humans and dragons lived together in harmony. But when an evil force threatened the land, the dragons sacrificed themselves to save humanity. Now, 500 years later, that same evil has returned and it’s up to a lone warrior, Raya, to track down the legendary last dragon to restore the fractured land and its divided people.",
		popularity: 2495.914,
		poster_path: "/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg",
		release_date: "2021-03-03",
		title: "Raya and the Last Dragon",
		video: false,
		vote_average: 8.3,
		vote_count: 2088,
	},
	{
		adult: false,
		backdrop_path: "/fev8UFNFFYsD5q7AcYS8LyTzqwl.jpg",
		genre_ids: [35, 10751, 16],
		id: 587807,
		original_language: "en",
		original_title: "Tom & Jerry",
		overview:
			"Tom the cat and Jerry the mouse get kicked out of their home and relocate to a fancy New York hotel, where a scrappy employee named Kayla will lose her job if she can’t evict Jerry before a high-class wedding at the hotel. Her solution? Hiring Tom to get rid of the pesky mouse.",
		popularity: 1689.341,
		poster_path: "/6KErczPBROQty7QoIsaa6wJYXZi.jpg",
		release_date: "2021-02-11",
		title: "Tom & Jerry",
		video: false,
		vote_average: 7.4,
		vote_count: 1130,
	},
];
