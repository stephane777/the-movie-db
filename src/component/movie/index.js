export { Movie } from "./movie.component";
