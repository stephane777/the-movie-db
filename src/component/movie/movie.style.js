import Card from "react-bootstrap/Card";
import styled from "styled-components";

export const CardWrapper = styled(Card)`
	border: 1px solid lightgray;
	border-radius: 8px;
	box-shadow: 0 2px 8px rgb(0 0 0 / 10%);
	& .card-img-top {
		border-top-left-radius: 8px;
		border-top-right-radius: 8px;
	}
`;
