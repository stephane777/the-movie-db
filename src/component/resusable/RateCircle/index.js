export { ratecircle } from "./ratecircle.component";
