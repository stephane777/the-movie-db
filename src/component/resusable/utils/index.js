export const formatDate = (dateToFormat) => {
	const options = { weekday: "short", year: "numeric", day: "numeric" };
	const newDateStr = new Date(dateToFormat).toLocaleString("en-GB", options);
	const index = newDateStr.length - 4;
	return `${newDateStr.slice(0, index)}, ${newDateStr.slice(index)}`;
};
