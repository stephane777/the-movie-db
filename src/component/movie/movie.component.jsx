import React from "react";
import { Card, Col } from "react-bootstrap";
import { RateCircle } from "../resusable/RateCircle/ratecircle.component";
import * as SM from "./movie.style";
import { formatDate } from "../resusable/utils";
import PropTypes from "prop-types";
const Movie = ({
	data: {
		poster_path: file_path,
		release_date,
		vote_average,
		original_title,
		id,
	},
}) => {
	const base_url = `https://image.tmdb.org/t/p`;
	const size = `w500`;

	const formattedDate = formatDate(release_date);

	return (
		<SM.CardWrapper
			as={Col}
			xs="8"
			sm="4"
			md="4"
			lg="3"
			xl="2"
			className="m-3 p-0"
		>
			<Card.Img
				variant="top"
				src={`${base_url}/${size}${file_path}`}
				alt={original_title}
			/>
			<RateCircle id={id} rate={vote_average}>
				{`${vote_average * 10}%`}
			</RateCircle>
			<Card.Body style={{ transform: "translateY(-25px)" }} className="py-0">
				<Card.Title className="mb-0">
					<strong
						style={{
							fontFamily: "'Source Sans Pro', Arial, sansSerif",
							fontSize: "1.6rem",
							fontWeight: 700,
						}}
					>
						{original_title}
					</strong>
				</Card.Title>
				<Card.Text style={{ color: "var(--gray)" }}>{formattedDate}</Card.Text>
			</Card.Body>
		</SM.CardWrapper>
	);
};

Movie.propTypes = {
	data: PropTypes.shape({
		poster_path: PropTypes.string,
		release_date: PropTypes.string.isRequired,
		vote_average: PropTypes.number.isRequired,
		original_title: PropTypes.string.isRequired,
		id: PropTypes.number.isRequired,
	}),
};
export { Movie };
