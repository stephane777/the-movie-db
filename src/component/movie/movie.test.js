import React from "react";
import "@testing-library/jest-dom";

import { render, screen, act, cleanup } from "@testing-library/react";
import { Movie } from "./movie.component";
import { resultMockAPI } from "../app/__test__/mockAPI";
import { formatDate } from "../resusable/utils";

const { original_title, file_path, release_date } = resultMockAPI[0];
const movie = resultMockAPI[0];

describe("it should render the component with data and img", () => {
	it("should render the Movie component", () => {
		const { container } = render(<Movie data={movie} />);
		expect(container.firstChild.className).toMatch(
			/sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8/
		);
	});
	it("should display an image", () => {
		render(<Movie data={movie} />);
		const img = screen.getByRole("img");
		expect(img).toBeInTheDocument();
		expect(img).toHaveAttribute("alt", original_title);
		expect(img).toHaveAttribute("src", file_path);
	});
	it("should display the title", () => {
		render(<Movie data={movie} />);
		expect(screen.getByText(original_title)).toBeInTheDocument();
	});
	it("should display the release date formatted", () => {
		render(<Movie data={movie} />);
		const formattedDate = formatDate(release_date);
		expect(screen.getByText(formattedDate)).toBeInTheDocument();
	});
});
describe("it should match snapshot", () => {
	it("should match the snapshot", () => {
		const testedUnit = render(<Movie data={movie} />);
		expect(testedUnit).toMatchSnapshot();
	});
});
