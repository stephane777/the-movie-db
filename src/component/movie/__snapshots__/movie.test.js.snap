// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
      >
        <img
          alt="Godzilla vs. Kong"
          class="card-img-top"
          src="https://image.tmdb.org/t/p/w500/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
        />
        <div
          class="sc-bdvvaa fzHxCj"
        >
          <canvas
            height="60"
            id="399566"
            style="background-color: transparent;"
            width="60"
          >
            85%
          </canvas>
        </div>
        <div
          class="py-0 card-body"
          style="transform: translateY(-25px);"
        >
          <div
            class="mb-0 card-title h5"
          >
            <strong
              style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
            >
              Godzilla vs. Kong
            </strong>
          </div>
          <p
            class="card-text"
          >
            24 Wed , 2021
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsDJrp ldqsHQ m-3 p-0 col-xl-2 col-lg-3 col-md-4 col-sm-4 col-8"
    >
      <img
        alt="Godzilla vs. Kong"
        class="card-img-top"
        src="https://image.tmdb.org/t/p/w500/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
      />
      <div
        class="sc-bdvvaa fzHxCj"
      >
        <canvas
          height="60"
          id="399566"
          style="background-color: transparent;"
          width="60"
        >
          85%
        </canvas>
      </div>
      <div
        class="py-0 card-body"
        style="transform: translateY(-25px);"
      >
        <div
          class="mb-0 card-title h5"
        >
          <strong
            style="font-family: 'Source Sans Pro', Arial, sansSerif; font-size: 1.6rem; font-weight: 700;"
          >
            Godzilla vs. Kong
          </strong>
        </div>
        <p
          class="card-text"
        >
          24 Wed , 2021
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
